src/main/java/frc/robot/subsystems/climber/Climber.java

```diff
@@ -0,0 +1,119 @@
+import frc.robot.subsystems.UnitModel;
+
+public class Climber extends SubsystemBase {
+    private final TalonFX master = new TalonFX(Ports.Climber.MASTER);
+    private final TalonFX slave = new TalonFX(Ports.Climber.SLAVE);
+    private final UnitModel unitModel = new UnitModel(Constants.Climber.TICKS_PER_METER);
+    private final DoubleSolenoid stopper = new DoubleSolenoid(Ports.Climber.STOPPER_FORWARD_CHANNEL, Ports.Climber.STOPPER_REVERSE_CHANNEL);
+    private final DoubleSolenoid gearboxShifter = new DoubleSolenoid(Ports.Climber.GEARBOX_FORWARD_CHANNEL, Ports.Climber.GEARBOX_REVERSE_CHANNEL);
+
+    public Climber() {
+        slave.follow(master);
+
+        master.setInverted(Ports.Climber.IS_MASTER_INVERTED);
+        slave.setInverted(Ports.Climber.IS_SLAVE_INVERTED);
+
+        master.setSensorPhase(Ports.Climber.IS_SENSOR_PHASE_INVERTED);
+
+        master.configMotionCruiseVelocity(Constants.Climber.CRUISE_VELOCITY, Constants.TALON_TIMEOUT);
+        master.configMotionAcceleration(Constants.Climber.ACCELERATION, Constants.TALON_TIMEOUT);
+
+        master.config_kP(0, Constants.Climber.KP, Constants.TALON_TIMEOUT);
+        master.config_kI(0, Constants.Climber.KI, Constants.TALON_TIMEOUT);
+        master.config_kD(0, Constants.Climber.KD, Constants.TALON_TIMEOUT);
+        master.config_kF(0, Constants.Climber.KF, Constants.TALON_TIMEOUT);
+    }
+
+    /**
+     * Get the climber's elevation relative to the ground.
+     *
+     * @return the climber's height [m].
+     */
+    public double getHeight() {
+        return unitModel.toUnits(master.getSelectedSensorPosition());
+    }
+
+    /**
+     * climb height meters up with the robot.
+     *
+     * @param height requested height to climb [m].
+     */
+    public void setHeight(double height) {
+        master.set(ControlMode.MotionMagic, unitModel.toTicks(height));
+    }
+
+    /**
+     * Set the stopper mode.
+     *
+     * @param mode the stopper mode.
+     */
+    public void setStopperMode(PistonMode mode) {
+        if (mode.getValue())
+            stopper.set(DoubleSolenoid.Value.kForward);
+        else
+            stopper.set(DoubleSolenoid.Value.kReverse);
+    }
+
+    /**
+     * Set the gearbox shifter mode to a given mode.
+     *
+     * @param mode the wanted gearbox shifter mode.
+     */
+    public void setGearboxMode(PistonMode mode) {
+        if (mode.getValue())
+            gearboxShifter.set(DoubleSolenoid.Value.kForward);
+        else
+            gearboxShifter.set(DoubleSolenoid.Value.kReverse);
+
+    }
+
+    /**
+     * Enum for piston mode, OPEN is true, CLOSED is false.
+     */
+    public enum PistonMode {
+        OPEN(true), CLOSED(false);
+        private boolean on;
+
+        PistonMode(boolean on) {
+            this.on = on;
+        }
+
+        /**
+         * Get the value of the piston mode.
+         *
+         * @return the value of the piston mode [boolean].
+         */
+        public boolean getValue() {

```

Comment:
Rename the function, it's hard to understand what is the "value", how to use it and why it's important